src/OrleansRuntime/Messaging/ObjectPool.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+namespace Orleans.Runtime
+{
+
+    /// <summary>
+    /// Simple object pool Interface.
+    /// Objects allocated should be returned to the pool when disposed.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal interface IObjectPool<T> where T : IDisposable
```

Comment:
Good point.  I was thinking we could pool the buffers later, but yeah, why pool only the buffers when we can pool the entire SAEA, which inclused the buffer.  Don't know where my brain was. :)